contracts/DCAStrategies/DCAStrategies/DCAStrategiesPositionsHandler.sol

```diff
@@ -56,7 +56,24 @@ abstract contract DCAStrategiesPositionsHandler is IDCAStrategiesPositionsHandle
   }
 
   /// @inheritdoc IDCAStrategiesPositionsHandler
-  function withdrawSwapped(uint256 _positionId, address _recipient) external override returns (uint256) {}
+  function withdrawSwapped(uint256 _positionId, address _recipient) external returns (TokenAmounts memory _tokenAmounts) {
+    if (!_isOwnerOf(_positionId, msg.sender)) revert IDCAStrategies.NotOwner();
```

Comment:
You should be checking if the `msg.sender` has permissions to execute a withdraw, by calling `hasPermission`. Otherwise, the permission system doesn't make much sense